src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -33,92 +30,36 @@ const useStyles = makeStyles(() => ({
-  const [filter, setFilter] = useState('');
+  const [filter, setFilter] = useState('post');
   const [fetchLoading, setFetchLoading] = useState(false);
-  const [searchEnabled, setSearchEnabled] = useState(false);
-  const [searchAuthors, { loading, data }] = useLazyQuery(SEARCH_QUERY, {
-    // Setting author variable here to use in the query above
-    variables: { author: searchText },
-    onCompleted: () => {
-      // We only want to display a feed once when searching posts for
-      // author, this would avoid the issue if there are duplicate
-      // feed results returned in our query
-      const finalResults = data.getPosts
-        .filter(
-          (element, index, self) =>
-            self.findIndex((result) => result.feed.id === element.feed.id) === index
-        )
-        .map((result) => {
-          return {
-            id: result.feed.id,
-            author: result.feed.author,
-            // The post will contain information about their most recent post to be used for AuthorResult component
-            post: {
-              title: result.title,
-              postLink: result.url,
-              postDate: new Intl.DateTimeFormat('en-US').format(result.published),
-            },
-          };
-        });
-      setResults({ type: 'author', searchResults: finalResults });
-    },
-  });
-
-  useEffect(() => {
-    async function searchPosts() {
-      // Setting to check length > 3, as the component will trying to fetch with 0 length causing error
-      if (searchText.length > 3) {
-        const encodedSearchText = encodeURIComponent(searchText);
-        try {
-          setFetchLoading(true);
-          const res = await fetch(`${telescopeUrl}/query?search=${encodedSearchText}`);
-          if (!res.ok) {
-            throw new Error(res.statusText);
-          }
-          const searchResults = await res.json();
-          // ES values property contains an array of objects with a (feed) id property
-          const posts = searchResults.values.map((result) => ({
-            id: result.id,
-            url: `/posts/${result.id}`,
-          }));
-          setFetchLoading(false);
-          setResults({ type: 'post', posts });
-        } catch (error) {
-          console.error('Something went wrong while fetching data', error);
-        }
-      } else {
-        setResults({ type: 'post', posts: [] });
-      }
-    }
 
-    if (filter === 'author') {
-      searchAuthors();
-    } else if (filter === 'post') {
-      searchPosts();
+  const search = async () => {
+    const encodedSearchText = encodeURIComponent(searchText);
+    try {
+      setFetchLoading(true);
+      const res = await fetch(`${telescopeUrl}/query?text=${encodedSearchText}&filter=${filter}`);
+      if (!res.ok) {
+        throw new Error(res.statusText);
+      }
+      const searchResults = await res.json();
+      // ES values property contains an array of objects with a (feed) id property
```

Comment:
This comment seems wrong, we pass  `post` ids, no?

Also, I think we should do this transform from `{id}` to `{id, url}` on the server, either in this PR or another.  Let's prepare the results to be useful right away from the REST api.  I actually thought we were doing this in another PR, which might still be in review?  cc @c3ho.

Reply:
I think we merged all new ES related PRs last week. I checked the [code](https://github.com/Seneca-CDOT/telescope/pull/1294/files) and we don't store `feedID` at least from what I see

Reply:
I'll change this. I mainly deleted what's not needed anymore but I didn't really pay much attention to the code that's going to stay.
Also, I think the backend already sends `{id, url}` (I think it's like this since the very beginning of telescope) it does it [here](https://github.com/Seneca-CDOT/telescope/blob/464dd741fbd8e4519da6b888ed63ddd1c0d81452/src/backend/web/routes/posts.js#L68), so I'll remove the transform (sorry about that, again I just deleted stuff without putting too much attention on what's going to stay).

Reply:
Awesome, thank you both for looking into this.  I'm reviewing this code again since we're touching it; sorry to make work for you @manekenpix.  We can file follow-ups on anything we find vs. delaying this surgery from being completed.

Reply:
My bad, the endpoint returning `{id, url}` is `GET /posts`. I just changed the indexer to return the same thing, and removed the transformation from the front-end.